irods/files/rule-bases/cyverse_logic.re

```diff
@@ -508,21 +509,23 @@ _cyverse_logic_ensureAVUEditable(*EditorName, *EditorZone, *Attr, *Val, *Unit) {
 	}
 }
 
-# If an AVU is not protected, it sets the AVU to the given item
-_cyverse_logic_setAVUIfUnprotected(*EntityType, *EntityName, *Attr, *Val, *Unit) {
-	if (!_cyverse_logic_isAVUProtected(*Attr)) {
+# If an AVU is allowed, it sets the AVU to the given item
+_cyverse_logic_setAVU(*EntityType, *EntityName, *Attr, *Val, *Unit) {
+	if (!_cyverse_logic_isAVUProtected(*Attr) || (_cyverse_logic_isAdm($userNameClient, $rodsZoneClient) && !_cyverse_logic_isAVUProtectedUUID(*Attr))) {
```

Comment:
iRODS is moving away from session variables. To make it easier for us to make the conversion, don't use them from rules or functions outside of the entrypoint rules for the rule base. Instead, pass them in as arguments.